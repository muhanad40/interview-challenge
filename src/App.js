import React from 'react';
import { Provider } from 'mobx-react';

import './App.css';
import Items from './Items';
import Preview from './Preview';
import SummaryHeader from './SummaryHeader';
import { ItemsModel } from './model';

const itemsStore = ItemsModel.create({
  items: [
    {
      id: 1002,
      name: 'Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots',
      dietaries: ['v', 've', 'rsfa'],
    },
    {
      id: 1003,
      name: 'Dill & Swiss Chard Potato Cakes, Summer Tabbouleh & Roasted Roots',
      dietaries: ['v', 've'],
    },
    {
      id: 1004,
      name: 'Hake & Smoky Chickpeas, Herby Potatoes & Turmeric Satay Broccoli',
      dietaries: ['rsf'],
    },
  ],
});

export const App = () => (
  <div className="wrapper">
    <SummaryHeader />

    <div className="container menu-builder">
      <div className="row">
        <Items />

        <Preview />
      </div>
    </div>
  </div>
);

export default () => (
  <Provider itemsStore={itemsStore}>
    <App />
  </Provider>
);
