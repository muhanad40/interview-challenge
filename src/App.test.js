require('./test-setup');

import React from 'react';
import App from './App';

import { shallow } from 'enzyme';

describe('App', () => {
  it('renders correctly', () => {
    const div = shallow(<App />);

    expect(div).toMatchSnapshot();
  });
});
