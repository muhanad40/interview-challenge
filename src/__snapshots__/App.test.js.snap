// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="wrapper"
>
  <SummaryHeader
    items={Array []}
  />
  <div
    className="container menu-builder"
  >
    <div
      className="row"
    >
      <Items
        items={Array []}
      />
      <Preview />
    </div>
  </div>
</div>
`;
